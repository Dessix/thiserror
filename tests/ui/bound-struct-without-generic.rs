use thiserror::Error;

#[derive(Error, Debug)]
#[error(bound = std::error::Error + 'static)]
struct BoundsWithoutGeneric {
    inner: u32,
}

impl std::fmt::Display for BoundsWithoutGeneric {
    fn fmt(&self, _formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
        unimplemented!()
    }
}

fn main() {}
