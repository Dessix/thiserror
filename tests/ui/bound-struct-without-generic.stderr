error: #[error(bound = ...)] requires at least one generic type parameter
 --> $DIR/bound-struct-without-generic.rs:4:1
  |
4 | #[error(bound = std::error::Error + 'static)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
